File: 3517-smallest-palindromic-rearrangement-i/3517-smallest-palindromic-rearrangement-i.java
class Solution {
    public String smallestPalindrome(String s) {
        int n = s.length();
        int half = n / 2;
        char[] chars = s.toCharArray();
        Arrays.sort(chars, 0, half); //half will be excluded bcs [0, half)
        for(int i = 0; i < half; i++) {
            chars[n - 1 - i] = chars[i];
        }

        return new String(chars);
    }
}

File: 3517-smallest-palindromic-rearrangement-i/3517-smallest-palindromic-rearrangement-i.py
class Solution:
    def smallestPalindrome(self, s: str) -> str:
        n = len(s)
        chars = list(s)
        half = sorted(chars[:n//2])
        ans = half[:]
        if n % 2:
            ans.append(chars[n//2]) #with middle char

        ans.extend(reversed(half))

        return "".join(ans)
